rom/dev/tests/helpers.rs

```diff
@@ -27,3 +29,35 @@ pub fn build_hw_model_and_image_bundle(
 
     (hw, image_bundle)
 }
+
+pub fn get_data(to_match: &str, haystack: &str) -> String {
+    let mut index = haystack.find(to_match).unwrap();
+    index += to_match.len();
+    let mut str = String::new();
```

Comment:
Could this be something like,

```
let end = haystack[index..].find("\n");
haystack[index..end]
```

Reply:
Looks like with the suggested change, the search terminates prematurely, returning the partial string. Will need to look more into this. We can take this as a separate improvement fix.

Reply:
Ack, could be an off-by-one error.